src/lib/fetch.ts

```diff
@@ -0,0 +1,41 @@
+import { TestBase } from "./testBase"
+import { ResourceTimingEntry } from "../@types"
+import { asyncGetEntry, normalizeEntry } from "./resourceTiming"
+
+export default class Fetch extends TestBase {
+    makeBeaconURL(): string {
+        return this.provider.makeFetchBeaconURL(this.config)
+    }
+
+    getResourceUrl(): string {
+        return this.provider.getResourceUrl(this.config)
+    }
+
+    makeTestSteps(): Promise<unknown[]> {
+        this.provider.markTestStart(this.config)
+        return Promise.all<unknown>([
+            this.test(),
+            this.provider.makeClientInfoPromise(this),
+        ])
+    }
+
+    test(): Promise<ResourceTimingEntry> {
+        return Promise.all<string, ResourceTimingEntry>([
+            this.fetchObjectAndId(),
+            asyncGetEntry(this.getResourceUrl(), 5000)
+        ])
+            .then(
+                ([id, entry]): ResourceTimingEntry => {
+                    const timing = normalizeEntry(entry)
+                    return this.provider.createFetchResult(timing, id, this.config)
+                }
+            )
+    }
+
+    fetchObjectAndId(): Promise<string> {
+        return fetch(this.getResourceUrl())
+            .then((res): string => {
+                return this.provider.getFetchHeaders(res.headers, this.config)
+            })
+    }
```

Comment:
`fetchObjectAndId()` is a historic detail to the Fastly provider implementation, specifically returning an ID string from a set of headers and may not not be useful in the core library? Maybe we could return the response object and pass that to `createFetchResult()` on line 30 above? This would allow providers to implement whatever logic they want to acquire an ID instead of just `getFetchHeaders`?
```suggestion
    fetchObject(): Promise<string> {
        return fetch(this.getResourceUrl())
    }
```

Reply:
Coming from Cedexis where this feature was always a big deal, it seemed natural to include in the base implementation, but I think your idea is much nicer.